utils/synthetics/getMarketsInfo.ts

```diff
@@ -55,9 +57,13 @@ export async function getMarketsInfo(
           longTokenInfo,
           shortTokenInfo,
           type: isSpotMarket ? 'Spot' : 'Perpetual',
-        }
+        } as MarketInfo
       })
-      .filter(Boolean) as MarketInfo[]
+    )
+
+    // Filtering out for any null results in market token info
+    return detailedMarketInfos.filter(Boolean) as MarketInfo[];
```

Comment:
i think you can do something like
```
`filter((m): m is MarketInfo => Boolean(m))` and there will be no need in type assertion (i.e. using "as")
```
it would be a bit cleaner